src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -5552,5 +6116,2874 @@ static void Test(params System.Collections.Generic.IEnumerable<long> a)
+                01 00 00 00
+            )
+
+        .maxstack 8
+
+        IL_0000: ldstr ""Test1""
+        IL_0005: call void [mscorlib]System.Console::Write(string)
+        IL_000a: ret
+    }
+
+    .method public hidebysig static 
+        void Test2 (
+            int64[] a
+        ) cil managed 
+    {
+        .param [1]
+            .custom instance void [mscorlib]System.ParamArrayAttribute::.ctor() = (
+                01 00 00 00
+            )
+
+        .maxstack 8
+
+        IL_0000: ldstr ""Test2""
+        IL_0005: call void [mscorlib]System.Console::Write(string)
+        IL_000a: ret
+    }
+
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldarg.0
+        IL_0001: call instance void [mscorlib]System.Object::.ctor()
+        IL_0006: ret
+    }
+}
+
+.class public auto ansi sealed beforefieldinit System.Runtime.CompilerServices.ParamCollectionAttribute
+    extends [mscorlib]System.Attribute
+{
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        .maxstack 8
+
+        IL_0000: ldarg.0
+        IL_0001: call instance void [mscorlib]System.Attribute::.ctor()
+        IL_0006: ret
+    }
+}
+";
+
+            var comp = CreateCompilationWithIL("", il);
+
+            var test1 = comp.GetMember<MethodSymbol>("Params.Test1").Parameters.Last();
+            var test2 = comp.GetMember<MethodSymbol>("Params.Test2").Parameters.Last();
+
+            VerifyParamAttribute(test1, isParamCollection: true);
+            VerifyParamAttribute(test2, isParamArray: true);
+
+            Assert.Empty(test1.GetAttributes());
+            Assert.Empty(test2.GetAttributes());
+
+            VerifyParamAttribute(test1, isParamCollection: true);
+            VerifyParamAttribute(test2, isParamArray: true);
+
+            AssertEx.Equal("System.Runtime.CompilerServices.ParamCollectionAttribute", test1.GetCustomAttributesToEmit(null).Single().ToString());
+            AssertEx.Equal("System.ParamArrayAttribute", test2.GetCustomAttributesToEmit(null).Single().ToString());
+
+            comp = CreateCompilationWithIL("", il);
+
+            test1 = comp.GetMember<MethodSymbol>("Params.Test1").Parameters.Last();
+            test2 = comp.GetMember<MethodSymbol>("Params.Test2").Parameters.Last();
+
+            Assert.Empty(test1.GetAttributes());
+            Assert.Empty(test2.GetAttributes());
+
+            VerifyParamAttribute(test1, isParamCollection: true);
+            VerifyParamAttribute(test2, isParamArray: true);
+
+            AssertEx.Equal("System.Runtime.CompilerServices.ParamCollectionAttribute", test1.GetCustomAttributesToEmit(null).Single().ToString());
+            AssertEx.Equal("System.ParamArrayAttribute", test2.GetCustomAttributesToEmit(null).Single().ToString());
+
```

Comment:
> I'm not sure if I'm missing something, but this seems to be exactly the same as the first part of the test, minus 2 calls to `VerifyParamAttribute` on `test1` and `test2`?

Are you referring to the to different sets of asserts performed on "equal" compilations above?

Since ```PEParameterSymbol._lazyIsParams``` can be initialized by two different APIs, the ```IsParamArray/IsParamCollection``` properties and ```GetAttributes```, I am asserting that result doesn't depend on the order in which the APIs are used first.